server/src/storage/segment.rs

```diff
@@ -33,20 +33,20 @@ use crate::storage::data_point::DataPoint;
 use crate::storage::{BUILDER_CAPACITY, StorageEngine};
 use crate::types::{Timestamp, TimestampArray, TimestampBuilder, Value, ValueBuilder};
 
-/// Shared functionality between different types of uncompressed segments, such as segment builders
-/// and spilled segments.
+/// Shared functionality between different types of uncompressed segments, such as [`SegmentBuilder`]
+/// and [`SpilledSegment`].
 pub trait UncompressedSegment {
     fn get_record_batch(&mut self) -> Result<RecordBatch, ParquetError>;
 
     fn get_memory_size(&self) -> usize;
 
-    // Since both segment builders and spilled segments are present in the compression queue, both
+    // Since both SegmentBuilders and SpilledSegments are present in the queue of finished segments, both
```

Comment:
Looks like links and `` are missing for `SegmentBuilder` and `SpilledSegment`. As you mentioned in a [comment](https://github.com/ModelarData/ModelarDB-RS/pull/23#discussion_r944129790) on #23 we can either use \[\`SegmentBuilder\`\]s or \[\`SegmentBuilders\`\]\(SegmentBuilder\). Generally when I have written documentation or papers I have only highlighted the name and not the plural s, but if you prefer that both are highlighted as you stated in the [comment](https://github.com/ModelarData/ModelarDB-RS/pull/23#discussion_r944129790) then that is fine by me.

Reply:
The reason links were not added was because this was a normal comment and not a doc comment. I have changed it to a doc comment and added links that include the `s` in the link.